### DISCUSSION TITLE
fix: textbox behaviour in fluentsearch in when searching text

### DISCUSSION THREAD

**Discussion by supriya1586** [microsoft/fluentui-blazor#2454] [Q&A]
for Search interactive in fluent search.. search functionality works fine. But search textbox doesnt work properly. when typed faster it eats up letter or behaves in any strage way possible. 
the problem if i add breakpoints and debug method i.e. also when i typed very slow.. everything works fine.
Can anyone help please?

i also tried adding debounce
below is my code

```razor
 <FluentSearch @ref=searchProjects
               @bind-Value="searchValue"
               Placeholder="Search Projects"
               @oninput="onInputDebounced"
               @onchange="HandleClear" />

 protected override void OnInitialized()
 {
     onInputDebounced = DebounceEvent<ChangeEventArgs>(e => SearchValue = (string)e.Value, TimeSpan.FromSeconds(0.5));
     base.OnInitialized();
 }

 Action<T> DebounceEvent<T>(Action<T> action, TimeSpan interval)
 {
     return Debounce<T>(arg =>
     {
         InvokeAsync(() =>
         {
             action(arg);
             StateHasChanged();
         });
     }, interval);
 }

 // Debounce and Throttle can be moved to another class
 Action<T> Debounce<T>(Action<T> action, TimeSpan interval)
 {
     if (action == null) throw new ArgumentNullException(nameof(action));

     var last = 0;
     return arg =>
     {
         var current = System.Threading.Interlocked.Increment(ref last);
         Task.Delay(interval).ContinueWith(task =>
         {
             if (current == last)
             {
                 action(arg);
                 OnSearchAsync();
             }
         });
     };
 }

 private string? SearchValue
 {
     get => searchValue;
     set
     {
         if (value != searchValue)
         {
             searchValue = value;
             InvokeAsync(async () => await OnSearchAsync());
         }
     }
 }

private void HandleClear(ChangeEventArgs args)
{
    searchProjectResult = projects?.AsQueryable();
    StateHasChanged();
}
```

also , especially when textfield is bound by a object property.. textbox behaviour is similarly strange
  @foreach (var activity in activities)
  {

      <div class="grid-item">

          @if ((activity.isActivityNameAdded || (!string.IsNullOrEmpty(activity.Name))) && !activity.isActivityNameEdited)
          {
              <FluentAnchor Style="font-size:12px" Href="#" Appearance="Appearance.Stealth" OnClick="()=>HandleActivityNameClick(activity)">@activity.Name</FluentAnchor>

          }
          @if ((!activity.isActivityNameAdded && (string.IsNullOrEmpty(activity.Name))) || activity.isActivityNameEdited)
          {
              <FluentTextField Class="activity-list-max-width" Placeholder="Add Activity" @bind-Value="@activity.Name"
                               Immediate="true" ImmediateDelay="1000"
                               @onkeydown="@(e=> HandleKeyDown(e, @activity.MilestoneId, activity))" @oninput="@(e=> UpdateInputText(e,activity))"></FluentTextField>

          }

**Comment by thieng101** [comment]
Hi I think I run into the same issue with you. I use **FluentSearch** with _Interactive with debounce_ and displayed the result to **FluentDataGrid**, it doesn't work properly. For example, when I type fast, some of the inputs disappear then reappear. But when I switch to **FluentSearch** with _Interactive (without debounce),_ it works fine.
